src/Neo/VM/Helper.cs

```diff
@@ -54,18 +54,64 @@ public static ScriptBuilder CreateArray<T>(this ScriptBuilder builder, IReadOnly
         /// <param name="builder">The <see cref="ScriptBuilder"/> to be used.</param>
         /// <param name="map">The key/value pairs of the map.</param>
         /// <returns>The same instance as <paramref name="builder"/>.</returns>
-        public static ScriptBuilder CreateMap<TKey, TValue>(this ScriptBuilder builder, IEnumerable<KeyValuePair<TKey, TValue>> map = null)
+        public static ScriptBuilder CreateMap<TKey, TValue>(this ScriptBuilder builder, IEnumerable<KeyValuePair<TKey, TValue>> map)
+            where TKey : notnull
+            where TValue : notnull
         {
-            builder.Emit(OpCode.NEWMAP);
-            if (map != null)
-                foreach (var p in map)
-                {
-                    builder.Emit(OpCode.DUP);
-                    builder.EmitPush(p.Key);
-                    builder.EmitPush(p.Value);
-                    builder.Emit(OpCode.SETITEM);
-                }
-            return builder;
+            var count = map.Count();
+
+            if (count == 0)
+                return builder.Emit(OpCode.NEWMAP);
+
+            foreach (var (key, value) in map.Reverse())
```

Comment:
Does we need the reverse?

Reply:
yes

Reply:
But it's more optimized if we change the tests, isn't it?

Reply:
tests are fine

Reply:
I know, but the test should be changed if fail, I think that it's faster to do it without this reverse

Reply:
I guess the expectation is that we have the same order (even though "order" doesn't make much sense for a map in general) of elements in the VM map as we have in a map passed to `CreateMap()`. Which means reversing elements either in VM (suboptimal) or here (much cheaper).

Reply:
But we need to preserve the order in a map?